packages/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -38,6 +38,27 @@ public function __construct( $declarations = array() ) {
 			return;
 		}
 		$this->add_declarations( $declarations );
+
+		// @TODO: This should be removed when and if safecss_filter_attr() in wp-includes/kses.php allows CSS custom properties.
+		if ( ! has_filter( 'safe_style_css', array( __CLASS__, 'safe_style_css_allow_wp_custom_property_filter' ) ) ) {
+			add_filter( 'safe_style_css', array( __CLASS__, 'safe_style_css_allow_wp_custom_property_filter' ), 10, 1 );
+		}
```

Comment:
I can't remember where I saw the discussion, but from memory one of the reasons custom properties aren't allowed for `safe_style_css` is that updating the custom property can be a way that someone without the `unfiltered_html` capability can update styles outside of the markup that they're generating. For example, someone could author a post that updates colors or values of a site's design when rendered on the frontend, rather than just the styles within the post that they've created. I'm not sure how valid that argument is, but that's what I could remember.

Reply:
Thanks for the explanation! 

I had a poke around for related discussions, but couldn't come up with much. 

It'd be great to track this reasoning down.